docker/plugin/digitile-monstache-plugin.go

```diff
@@ -0,0 +1,459 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/rwynn/monstache/monstachemap"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"log"
+	"os"
+	"strings"
+	"sync"
+)
+
+var warnLog = log.New(os.Stdout, "WARN ", log.Flags())
+
+var assetFields = map[string]bool{
+	"_id":             true,
+	"org":             true,
+	"filename":        true,
+	"filetype":        true,
+	"content":         true,
+	"tags":            true,
+	"thumbnail_url":   true,
+	"remote":          true,
+	"users":           true,
+	"created":         true,
+	"remote_modified": true,
+	"owner":           true,
+	"editors":         true,
+	"counters":        true,
+}
+
+var searchUpdateFields = map[string]bool{
+	"filename":        true,
+	"content":         true,
+	"tags":            true,
+	"thumbnail_url":   true,
+	"users":           true,
+	"remote_modified": true,
+	"editors":         true,
+	"counters":        true,
+	"forceSync":       true,
+}
```

Comment:
Why are only a subset of the fields being watched? If file type was to change, it doesn't re-sync the asset correct?

Reply:
This functionality and list of fields is not new but was reviewed last week. Changes to fields  not included here will not sync the asset. If the filetype was ever to change then the forceSync approach can be used. @jtopel what are your thoughts on this?

Reply:
@inventa-consulting This is our fault since we reviewed and approved this last week. After discussing the process the filetype will need to be included here. Here is the example -  When we first fetch metadata from Google Drive files will be given an unknown file type as we begin to process and enhance the asset the filetype will be set. As it pertains to ES- the filetype will change in transit.

Reply:
@inventa-consulting another field that I think was overlooked her is the owner field. We discussed it, but I think it slipped thru and then we missed it in review. I have attached an example of asset JSON here with the owner field at the bottom:
`{
    "_id" : ObjectId("5c6afd14808cc60ce4e6fd6e"),
    "org" : ObjectId("5ac4fbf8737e93031eadacaa"),
    "name" : " Zamzar Images Analysis",
    "filename" : " Zamzar Images Analysis",
    "filetype" : ObjectId("5c3c238a1969064b6812d0aa"),
    "content" : "",
    "tags" : [ 
        ObjectId("5cae1f45808cc619b5e5bbb9"), 
        ObjectId("5ea707d224621300010fbbae"), 
        ObjectId("5cae1f33808cc619b5e5bb97"), 
        ObjectId("5cae1f45808cc619b5e5bbb9"), 
        ObjectId("5d674ea6808cc642752f032c"), 
        ObjectId("5dd23374808cc63e3e0d486e"), 
        ObjectId("5c7582b6013f673f4b6ab568"), 
        ObjectId("5cae1f45808cc619b5e5bbb9"), 
        ObjectId("5d674ea6808cc642752f032c"), 
        ObjectId("5cae1f45808cc619b5e5bbb9"), 
        ObjectId("5d66b31a808cc661a8750b3c"), 
        ObjectId("5ee7dbbcd4ff4400019a810a"), 
        ObjectId("5ee7dda7d4ff4400019a810f")
    ],
    "folders" : [ 
        ObjectId("5ac5b486288ee77a338a25b8")
    ],
    "scores" : {},
    "bookmarks" : [],
    "screenshots" : [],
    "screenshot_urls" : [],
    "helpful" : [],
    "ratings" : {},
    "counters" : {},
    "dimensions" : {},
    "performance" : {
        "thumbnail" : 1.63482928276062,
        "screenshot" : 0.255759000778198
    },
    "created" : ISODate("2018-06-21T18:42:20.546Z"),
    "modified" : ISODate("2019-11-07T21:15:16.659Z"),
    "remote_modified" : ISODate("2018-06-22T14:59:29.160Z"),
    "remote" : {
        "mime_type" : "application/vnd.google-apps.spreadsheet",
        "id" : "1OBwm57zt1zabehtdLWaG2bV8aw2NjuNhuJzsVa2p2qE",
        "download_url" : null,
        "owner" : {
            "address" : "jtopel@digitile.io",
            "photo_link" : "https://lh3.googleusercontent.com/a-/AOh14GhHgP4DYmfl3NmhellvupPfo5v41FZpGHep7_Vq=s64",
            "name" : "Josh Topel"
        },
        "drive_id" : null,
        "source" : "google_drive",
        "bytes" : null
    },
    "users" : [ 
        "meichner@digitile.io", 
        "jtopel@digitile.io"
    ],
    "editors" : [ 
        "meichner@digitile.io", 
        "jtopel@digitile.io"
    ],
    "downloaders" : [ 
        "jtopel@digitile.io"
    ],
    "deleters" : [ 
        "jtopel@digitile.io"
    ],
    "renamers" : [ 
        "meichner@digitile.io", 
        "jtopel@digitile.io"
    ],
    "sharers" : [ 
        "meichner@digitile.io", 
        "jtopel@digitile.io"
    ],
    "recipients" : {
        "shares" : [ 
            {
                "email_id" : "snigdhakadam23@gmail.com",
                "name" : "snigdha kadam",
                "permission_id" : "00007856385651401853",
                "photo_link" : null,
                "permission" : "editor"
            }, 
            {
                "email_id" : "meichner@digitile.io",
                "name" : "Michelle Eichner",
                "permission_id" : "07024776923912268397",
                "photo_link" : "https://lh3.googleusercontent.com/a-/AOh14GiOOpyl5E2K-sYEbrf4Z9o7Q5Owh-3qbRoMi5pUpg=s64",
                "permission" : "editor"
            }, 
            {
                "name" : "Digitile",
                "permission_id" : "07470220578039249076k",
                "permission" : "viewer"
            }, 
            {
                "email_id" : "jtopel@digitile.io",
                "name" : "Josh Topel",
                "permission_id" : "03305360068109155544",
                "photo_link" : "https://lh3.googleusercontent.com/a-/AOh14GhHgP4DYmfl3NmhellvupPfo5v41FZpGHep7_Vq=s64",
                "permission" : "owner"
            }
        ]
    },
    "status" : {},
    "tried_processing" : true,
    "url" : "https://docs.google.com/spreadsheets/d/1OBwm57zt1zabehtdLWaG2bV8aw2NjuNhuJzsVa2p2qE/edit?usp=drivesdk",
    "version" : "30",
    "name_content" : "zamzar images analysis  zamzar images analysis",
    "tag_content" : "marketing approved my drive marketing digitile demo folder demo marketing digitile",
    "expiration" : ISODate("2019-04-12T15:52:21.493Z"),
    "description" : "",
    "owner" : "jtopel@digitile.io"
}`

Reply:
I added owner and filetype to the list of updatable fields